packages/edit-site/src/components/add-new-template/utils.js

```diff
@@ -110,6 +112,24 @@ export const entitiesConfig = {
 	taxonomy: {
 		...taxonomyBaseConfig,
 		templatePrefix: 'taxonomy-',
+		getTitle: ( labels, slug ) => {
+			const _needsUniqueIdentifier = needsUniqueIdentifier(
+				labels,
+				slug
+			);
+			return sprintf(
+				// translators: %1s: Name of the taxonomy e.g: "Category"; %2s: Slug of the taxonomy e.g: "product-categories".
+				__( 'Taxonomy: %1$s %2$s' ),
+				labels.singular_name,
+				_needsUniqueIdentifier ? `(${ slug })` : ''
```

Comment:
Should probably leave the punctuation up to translators:

```suggestion
				__( 'Taxonomy: %1$s (%2$s)' ),
				labels.singular_name,
				slug
```

Which of course means moving the ternary out of the sprintf call, and instead have two separate sprintf calls.